### ISSUE TITLE
Automatic linkification of URLs is broken when URL contains a '@' character

### ISSUE THREAD

**Issue by berrange** [quotient-im/Quaternion#850]
### Description

In one of the channels I'm in, someone posted a message as follows

```
bonzini are you going to pick up https://patchew.org/QEMU/TY0PR0101MB4285923FBE9AD97CE832D95BA4E59@TY0PR0101MB4285.apcprd01.prod.exchangelabs.com/?
```

Quaternion linkified the URL incorrectly, turning it into two separate links

 * A link to ``https://patchew.org/QEMU/``
 * A link to ``mailto:TY0PR0101MB4285923FBE9AD97CE832D95BA4E59@TY0PR0101MB4285.apcprd01.prod.exchangelabs.com``

Obviously, I was expecting it to link to the original URL as typed.

I guess it is looking for certain text patterns and the mailto pattern detection is running at higher priority than the URL pattern, or perhaps the URL pattern is not permitting the "@" character ?

In case it is at all relevant, the raw message content was this:

```
{
    "content": {
        "body": "bonzini are you going to pick up https://patchew.org/QEMU/TY0PR0101MB4285923FBE9AD97CE832D95BA4E59@TY0PR0101MB4285.apcprd01.prod.exchangelabs.com/?",
        "msgtype": "m.text"
    },
    "event_id": "$167646434940023uzeKm:matrix.org",
    "origin_server_ts": 1676464349968,
    "sender": "@_oftc_stsquad:matrix.org",
    "type": "m.room.message",
    "unsigned": {
        "age": 2706
    }
}
```

And this channel is ``#qemu`` on the IRC server on ``oftc.net``, that is bridged from the ``matrix.org`` server

### Steps to reproduce

- Send the message shown above
- Mouse-over the URL to see what link it points to, and observe it points to two different links depending on where in the URL you position the mouse

### Version information

- **Quaternion version**: quaternion-0.0.95.1-3.fc37.x86_64
- **Qt version**: qt5-qtbase-5.15.8-2.fc37.x86_64
- **Install method**: Fedora 37 RPM packages
- **Platform**: Fedora 37, x86_64 host, running under a regular GNOME desktop session

**Comment by KitsuneRal** [comment]
Indeed, the e-mail regex is applied first and matches things with `@` too aggressively.
